src/main/java/net/snowflake/ingest/streaming/internal/StreamingIngestStage.java

```diff
@@ -226,7 +226,8 @@ synchronized SnowflakeFileTransferMetadataWithAge refreshSnowflakeMetadata(boole
     JsonNode responseNode = this.parseClientConfigureResponse(response);
     // Do not change the prefix everytime we have to refresh credentials
     if (Utils.isNullOrEmpty(this.clientPrefix)) {
-      this.clientPrefix = responseNode.get("prefix").textValue();
+      this.clientPrefix = responseNode.get("prefix").textValue() +
+              (responseNode.has("deployment_id") ? ("_" + responseNode.get("deployment_id").longValue()) : "");
```

Comment:
Ideally we should have an object on the response, but I guess we just didn't do it before :(

Reply:
Hehe if we had an object, this change would have been even trickier because I think that old clients would fail if they found a new field and a jackson exception. I think in that case, we would have to use some jackson annotation magic to tell the client to ignore the field if it exists and we do not expect it (new SF old client) and to put a default value if we expect it and it is not there (old SF new client).